docs/secret-rotation.md

```diff
@@ -0,0 +1,254 @@
+# Secret Rotations &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; **FluxCD**
+
+This brief guide should outline all the steps needed to be able to swap out a set of secrets or credentials with FluxCD GitOps.
+
+## [**Pre-requisites**](#pre-requisites-link)
+#### [Useful Resources](#useful-resources-link)
+#### [Applications & Software](#applications--software-link)
+
+## [**Outline**](#outline-link)
+#### [Cluster Maintenance](#cluster-maintenance-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Basics of what is done in a standard secret rotation.
+#### [Security Threat](#security-threat-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - What to do in the case of a compromised secret.
+
+## [**Horizon Application Credentials**](#horizon-application-credentials-link)
+#### [Create/Delete Credentials](#createdelete-credentials-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - How to create & delete application credentials within Horizon.
+
+## [**Processing Secrets**](#processing-secrets-link)
+#### [Encrypting Credentials](#encrypting-new-credential-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Using Kubeseal to encrypt the newly created credentials.
+#### [Updating Secrets](#updating-secrets-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - How to rotate new secrets using FluxCD’s GitOps.
+#### [Deleting Old Secrets](#deleting-old-secrets-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Some housekeeping steps post secret rotation.
+#### [Rotating Compromised Secrets](#rotating-compromised-secrets-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Summary for the rotation of secrets posing a security threat.
+
+## [**Validate Configuration**](#validate-configuration-link)
+#### [Sonobuoy](#sonobuoy-link)
+&nbsp; &nbsp; &nbsp; &nbsp; - Brief outline on validating the Kubernetes configuration.
+
+#
+<a id="pre-requisites-link"></a>
+# **Pre-Requisites**
+
+This document assumes that a self-managed CAPI cluster deployed with FluxCD is already configured and deployed, if not, one can be deployed by following the instructions found in [this](https://github.com/stackhpc/capi-helm-fluxcd-config) GitHub repository.
+
+In addition to this, the user trying to rotate the secrets should have an account on Horizon with the necessary permissions.
+
+<a id="useful-resources-link"></a>
+### **Useful Resources**
+
+[CAPI cluster FluxCD repository](https://github.com/stackhpc/capi-helm-fluxcd-config)
+
+[Waldur FluxCD Demo repository](https://github.com/stackhpc/waldur-fluxcd-gitops-demo)
+
+[CAPI Cluster debugging manual](https://github.com/azimuth-cloud/capi-helm-charts/blob/main/charts/openstack-cluster/DEBUGGING.md)
+
+<a id="applications--software-link"></a>
+### **Applications & Software**
+
+| This document assumes that the Kubernetes cluster’s ```kubeconfig``` is known. |
+| :---- |
```

Comment:
Can we replace these types of box formatting with something which looks better in GitHub-rendered markdown like these please https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Reply:
Also the `kubeconfig` should be in your cluster config repo at `clusters/<cluster-name/kubeconfig` so probably worth mentioning that.